### DISCUSSION TITLE
How to validate two select fields with same values.

### DISCUSSION THREAD

**Discussion by apoorvmote** [react-hook-form/react-hook-form#4724] [Q&A] [answered]
https://codesandbox.io/s/react-hook-form-watch-v7-ts-forked-0u78n?file=/src/index.tsx

In above form I am trying to transfer money from one account to another account. The list of accounts is same for both dropdown options. 

Basically how do I validate `from account` and `to account` are not same? 

Now user may choose `to account` first then choose `from account` or vice versa. User may change mind anytime. So which field I should show error or show error on both fields?

**Comment by jorisre** [comment, marked as answer]
Here is a way to achieve that: https://codesandbox.io/s/react-hook-form-watch-v7-ts-forked-mkhg8

1. Add `mode: "onChange"`
2. Validate field by checking value of the other field
3. Display an error message

**Reply by apoorvmote** [reply]
It does work but if I choose `to account` to `bank 1` and then set `from account` to `bank 1`. Then error is shown at `from account`. Now instead of updating `from account` I changed `to account` to `bank 2` then error message is still showing until I press the submit button.

**Reply by jorisre** [reply]
You should trigger the validation yourself
https://react-hook-form.com/api/useform/trigger

**Reply by apoorvmote** [reply]
Also the form is quite large and setting `mode: "onChange"` affects other parts of form also. And I will lose the benefit of uncontrolled form by rendering on each key stroke.

**Reply by bluebill1049** [reply]
no, you are not. hook form only re-render when error state gets updated. we don't trigger re-render on user's typing.
